[Multi-bucket] Storage - add multi bucket setup and usage examples (#7858)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/set-up-storage/index.mdx

});
```

### Configure additional storage buckets
Amplify Storage allows you to optionally configure your backend to automatically create and manage multiple storage resources. 

Start by creating additional storage resources in your backend using the `defineStorage` function and providing a `name` for your storage bucket. The `name` attribute is required and we recommend assigning a developer friendly value for each storage bucket to allow you to easily reference them from your application code later. You can define any custom access controls here as well.

> Note that if you define additional storage buckets one of them must be marked with `isDefault: true`.

Reviewer: ```suggestion
Note that if you define additional storage buckets one of them must be marked as default with `isDefault` flag..
```

```ts title="amplify/storage/resource.ts"
export const primaryStorage = defineStorage({
  name: 'primaryStorage',
  isDefault: true, // identify your default storage bucket (required)
});

export const secondaryStorage = defineStorage({
  name: 'secondaryStorage',
  access: (allow) => ({
    'private/{entity_id}/*': [allow.entity('identity').to(['read', 'write', 'delete'])]
  })
})
```

Add your additional storage resources to your backend definition.

```ts title="amplify/backend.ts"
import { defineBackend } from '@aws-amplify/backend';
import { auth } from './auth/resource';
import { primaryStorage, secondaryStorage } from './storage/resource';

defineBackend({
  auth,
  primaryStorage,
  secondaryStorage
});
```

> Additional storage buckets can easily be referenced from your application code in Amplify Storage by passing the bucket option. The `bucket` option can either be a string representing the target bucket's assigned name in Amplify Backend or an object specifying the bucket name and region from the console as strings.

Reviewer: ```suggestion
Additional storage buckets can easily be referenced from your application code in Amplify Storage by passing the `bucket` option. You can reference the bucket either by providing the target bucket's assigned name in Amplify Backend or by specifying the actual bucket name and region from the console.
```

I suggest this, since not all libraries will have same type info and can refer specific type info in API section

See each Amplify Storage API page for additional usage examples.

```ts
import { downloadData } from 'aws-amplify/storage';

try {
  const result = downloadData({
    path: "album/2024/1.jpg",
    options: {
    bucket?: "secondaryStorage" // Specify a target bucket using name assigned in Amplify Backend
    // Alternatively, provide bucket name from console and associated region
    // bucket?: { bucketName: "bucket-name-from-console", region: "us-east-2" }
    }
  }).result;
} catch (error) {
  console.log(`Error: ${error}`)
}
```

### Use existing storage resources without Amplify Backend

While using the Amplify Backend is the easiest way to get started, existing storage resources can be integrated with Amplify Storage. 

You can accomplish this by passing your existing storage resource metadata to `Amplify.configure`. This will configure the Amplify Storage client library to interact with your existing storage resources. It's recommended to add the Amplify configuration step to your app's root entry point.

```ts
import { Amplify } from 'aws-amplify';

Amplify.configure({
  Storage: {
    S3: {

>       bucket: "<your-primary-bucket-name>",

Reviewer: lets add a comment here suggesting this is default 
```suggestion
      // default bucket name
      bucket: "<your-primary-bucket-name>",
```

>       region: "<your-primary-bucket-region>",

Reviewer: ```suggestion
      // default bucket region
      region: "<your-primary-bucket-region>",
```

    	// your default bucket metadata should be duplicated below
      buckets: {
        "<your-primary-bucket-friendly-name>": {
          bucketName: "<your-primary-bucket-name>",
          region: "<your-primary-bucket-region>",
			},
        "<your-additional-bucket-friendly-name>": {
          bucketName: "<your-additional-bucket-name>",
          region: "<your-additional-bucket-region>",
			},
      },
    },
  },
});
```

> Alternatively, you can use existing storage resources in your app by creating or modifying the `amplify_outputs.json` file directly. Import the file into your application and pass it the `Amplify.configure` function to configure the Amplify Storage client library.  

Reviewer: ```suggestion
Alternatively, you can use existing storage resources in your app by creating or modifying the `amplify_outputs.json` file directly. Read more about [amplify_outputs.json](/[platform]/reference/amplify_outputs/).
```

Read more about [amplify_outputs.json](/[platform]/reference/amplify_outputs/).

```ts title="amplify_outputs.json"
{
  // ...
  "storage": {
    // default bucket region
    "aws_region": "us-east-2", 

>     // default bucket name should match the AWS console

Reviewer: ```suggestion
    // default bucket name
```
